sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubAsyncProducer.java

```diff
@@ -296,19 +305,46 @@
         verifyPartitionKey(partitionKey);
 
         return sendLinkMono.flatMap(link -> {
+            final AtomicReference<Context> sendSpanContext = new AtomicReference<>(Context.NONE);
             return link.getLinkSize()
                 .flatMap(size -> {
                     final int batchSize = size > 0 ? size : MAX_MESSAGE_LENGTH_BYTES;
                     final BatchOptions batchOptions = new BatchOptions()
                         .partitionKey(partitionKey)
                         .maximumSizeInBytes(batchSize);
 
-                    return events.collect(new EventDataCollector(batchOptions, 1, () -> link.getErrorContext()));
+                    return events.map(eventData -> {
+                        Context parentContext = eventData.context();
+                        Context entityContext = parentContext.addData(entityPath, link.getEntityPath());
+                        sendSpanContext.set(TraceUtil.start("send", entityContext.addData(hostName, link.getHostname())));
```

Comment:
Using the static method to `start` will be hard to unit test.

Reply:
I'm hoping doing something more DI-ish like this will remove the need for it.
https://github.com/Azure/azure-sdk-for-java/pull/5061#discussion_r316944552